lib/Cryptol.cry

```diff
@@ -323,3 +324,133 @@ groupBy : {each,parts,elem} (fin each) =>
   [parts * each] elem -> [parts][each]elem
 groupBy = split`{parts=parts}
 
+/**
+ * Logical implication
+ */
+(==>) : {a} a -> a -> a
+a ==> b = not a || b
```

Comment:
Weren't we going to specialize this to type Bit? I would actually prefer to redefine it as

    (==>) : Bit -> Bit -> Bit
    a ==> b = if a then b else True

to make it explicitly lazy in the second argument. (According to the Cryptol semantics that hasn't been written yet, I think bitwise operators like `(||)` will be strict in both arguments.) This will be important for using `(==>)` to state properties with preconditions that guard against undefined behavior.